pkg/clusteroperator/clusteroperators.go

```diff
@@ -0,0 +1,273 @@
+				}
+			}
+
+			// Here we are always returning "false, nil" so we keep iterating throughout the stableInterval
+			// of the inner wait.pollImmediate loop, until we time out.
+			return false, nil
+		})
+
+		if isAllCoAvailable {
+			glog.V(100).Infof("All ClusterOperators were found available during availableDuration: %v",
+				availableDuration)
+
+			return true, nil
+		}
+
+		glog.V(100).Infof("Not all ClusterOperators were found available during availableDuration: %v, "+
+			"retrying ...", availableDuration)
+
+		// keep iterating in the outer wait.PollImmediate waiting for cluster to be stable.
+		return false, nil
+
+	})
+
+	if err == nil {
+		glog.V(100).Infof("All ClusterOperators were found available  during availableDuration: %v",
+			availableDuration)
+	} else {
+		// Here err is "timed out waiting for the condition"
+		glog.V(100).Infof("Not all ClusterOperators were found available  during availableDuration: %v",
+			availableDuration)
+	}
+
+	return false, err
+}
+
+// WaitForAllStopProgressing waits until all clusteroperators will finish with a progressing (PROGRESSING=False).
+func (builder *CoListBuilder) WaitForAllStopProgressing(availableDuration time.Duration, timeout time.Duration) error {
+	glog.V(100).Infof("Wait for all ClusterOperators from the ClusterOperatorsList ended "+
+		"with a Progressing up to duration of %v", timeout)
+
+	isAllCoEndProgressing := true
+
+	// Wait 5 secs in each iteration before condition function () returns true or errors or times out
+	// after availableDuration
+	err := wait.PollImmediate(fiveScds, timeout, func() (bool, error) {
+
+		isAllCoEndProgressing = true
+
+		// check if all clusteroperators end with progressing stage every 5 seconds
+		// during entire availableDuration time period
+		_ = wait.PollImmediate(fiveScds, availableDuration, func() (done bool, err error) {
+
+			err = builder.Discover()
+
+			if err != nil {
+				return false, err
+			}
+
+			// iterate through the ClusterOperators in the list.
+			for _, clusteroperator := range builder.ObjectList.Items {
+				for _, condition := range clusteroperator.Status.Conditions {
+					if condition.Type == "Progressing" && condition.Status == isTrue {
+						isAllCoEndProgressing = false
+					}
+
+					glog.V(100).Infof("ClusterOperators: %v still in Progressing State", clusteroperator.Name)
+				}
+			}
+
+			// Here we are always returning "false, nil" so we keep iterating throughout the stableInterval
+			// of the inner wait.pollImmediate loop, until we time out.
+			return false, nil
+		})
+
+		if isAllCoEndProgressing {
+			glog.V(100).Infof("All ClusterOperators ended with a Progressing Stage "+
+				"during availableDuration: %v", availableDuration)
+
+			return true, nil
+		}
+
+		glog.V(100).Infof("Not all ClusterOperators ended with a Progressing Stage "+
+			"during availableDuration: %v, retrying ...", availableDuration)
+
+		// keep iterating in the outer wait.PollImmediate waiting for cluster to be stable.
```

Comment:
Please adjust comment to your function. This comment from mcpList.go

Reply:
done